internal/nodelabels/nodelabels_test.go

```diff
@@ -0,0 +1,282 @@
+package nodelabels
+
+import (
+	"context"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/neilotoole/slogt"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func TestSanitizeLabelName(t *testing.T) {
+	tests := []struct {
+		name         string
+		input        string
+		expected     string
+		checkExact   bool
+		checkPrefix  bool
+		prefixExpect string
+	}{
+		{
+			name:       "simple name",
+			input:      "my-images",
+			expected:   "my-images",
+			checkExact: true,
+		},
+		{
+			name:       "with underscores",
+			input:      "my_images",
+			expected:   "my_images",
+			checkExact: true,
+		},
+		{
+			name:       "with dots",
+			input:      "my.images",
+			expected:   "my.images",
+			checkExact: true,
+		},
+		{
+			name:       "with spaces (invalid)",
+			input:      "my images",
+			expected:   "my-images",
+			checkExact: true,
+		},
+		{
+			name:       "starts with dash (invalid)",
+			input:      "-my-images",
+			expected:   "my-images",
+			checkExact: true,
+		},
+		{
+			name:       "ends with dash (invalid)",
+			input:      "my-images-",
+			expected:   "my-images",
+			checkExact: true,
+		},
+		{
+			name:         "too long",
+			input:        "this-is-a-very-long-instance-name-that-exceeds-sixty-three-characters-and-should-be-truncated",
+			checkPrefix:  true,
+			prefixExpect: "this-is-a-very-long-instance-name",
+		},
+		{
+			name:       "special characters",
+			input:      "my@images!",
+			expected:   "my-images",
+			checkExact: true,
+		},
+		{
+			name:       "empty string",
+			input:      "",
+			expected:   "prefetcher",
+			checkExact: true,
+		},
+		{
+			name:       "only invalid chars",
+			input:      "!!!",
+			expected:   "prefetcher",
+			checkExact: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			output := sanitizeLabelName(tt.input)
+
+			if tt.checkExact {
+				assert.Equal(t, tt.expected, output)
+			}
+			if tt.checkPrefix {
+				assert.True(t, strings.HasPrefix(output, tt.prefixExpect))
+			}
+
+			// Verify output is a valid label name
+			assert.LessOrEqual(t, len(output), MaxLabelNameLength, "label name too long")
+			assert.NotEmpty(t, output, "label name should not be empty")
+			if len(output) > 0 {
+				assert.Regexp(t, `^[a-zA-Z0-9]`, output, "must start with alphanumeric")
+				assert.Regexp(t, `[a-zA-Z0-9]$`, output, "must end with alphanumeric")
+			}
+		})
+	}
+}
+
+func TestLabelKeyConstruction(t *testing.T) {
+	instanceName := "my-images"
+	sanitized := sanitizeLabelName(instanceName)
+	labelKey := LabelPrefix + sanitized
+
+	assert.Equal(t, "image-prefetcher.stackrox.io/my-images", labelKey)
+	assert.True(t, strings.HasPrefix(labelKey, LabelPrefix))
+}
```

Comment:
I'd say this test is well beyond the https://www.goodreads.com/quotes/10376041-write-tests-until-fear-is-transformed-into-boredom bar. Drop it?